File: Aula_Ricardo/Programação_orientada_a_objetos/python/Composição.py
class Motor:
    def __init__(self):
        self.estado = "desligado"

    def ligar(self):
        self.estado = "ligado"
        return "Motor ligado"
    
    def desligar(self):
        self.estado = "desligado"
        return "Motor desligado"
    
class Carro:
    def __init__(self):
        self.motor = Motor()

    def ligar_carro(self):
        return self.motor.ligar()
    
    def desligar_carro(self):
        return self.motor.desligar()
    
if __name__ == "__main__":
    carro = Carro()
    print(carro.ligar_carro())
    print(carro.desligar_carro())

File: Aula_Ricardo/Programação_orientada_a_objetos/java/Composição.java
class Motor {
    private String estado;

    public Motor() {
        this.estado = "desligado";
    }

    public String ligar() {
        this.estado = "ligado";
        return "Motor ligado";
    }

    public String desligar() {
        this.estado = "desligado";
        return "Motor desligado";
    }
}

class Carro {
    private Motor motor;

    public Carro() {
        this.motor = new Motor();
    }

    public String ligarCarro() {
        return motor.ligar();
    }

    public String desligarCarro() {
        return motor.desligar();
    }
}

public class Main {
    public static void main(String[] args) {
        Carro carro = new Carro();
        System.out.println(carro.ligarCarro());
        System.out.println(carro.desligarCarro());
    }
}
